### ISSUE TITLE
I cannot filter the blocks by custom category

### ISSUE THREAD

**Issue by SergiArias** [Log1x/acf-composer#90]
I don't want all Gutenberg blocks to be available, so I use the following function:

```php
add_filter( 'allowed_block_types_all', function ($allowed_blocks) {
    // Get widget blocks and registered by plugins blocks
    $registered_blocks = \WP_Block_Type_Registry::get_instance()->get_all_registered();
    // var_dump($registered_blocks);
    // wp_die();

    // Now $registered_blocks contains only blocks registered by plugins, but we need keys only of acf ones
    $registered_blocks = array_filter(array_keys($registered_blocks), function($block) {
        return strpos($block, 'acf/') !== false;
    });

    // Merge allowed core blocks with plugins blocks
    return array_merge([
        // 'core/image',
        'core/paragraph',
        'core/separator',
        // 'core/heading',
        // 'core/list'
    ], $registered_blocks);
});
```

But taking a look at all registered blocks, all of the acf created blocks don't specify category (block category appears as null) although it appears correctly in the editor.

I would like to filter by category instead of "acf/" as I have now. Is there a way to do it? Is this "null" category situation a consequence of acf composer or is it an advanced custom fields issue?

**Comment by Log1x** [comment]
Were you ever able to figure this out?

**Comment by SergiArias** [comment]
@Log1x not really, as just filtering the way I did worked although not being the way it should.

But I just opened an issue now, let's see what they say: AdvancedCustomFields/acf#596

It's an ACF issue clearly, so we can close the issue I think.

**Comment by Log1x** [comment]
Are you registering the block categories using Poet?

https://github.com/Log1x/poet/blob/master/src/Poet.php#L68-L76

Could you try adjusting this to after_setup_theme or adding a different priority?

This is in response to the reply on your issue.

**Comment by SergiArias** [comment]
I usually do, but in this case I did the example just using acf, and not using a custom category (I used "formatting" which is a wp one) to better isolate the problem.

Anyway I tried to change the priority/hook but nothing changed. I think ACF is filling the category property after the hook "allowed_block_types_all" and therefore it's not available for filtering (the wp not-ACF blocks are showing the category in this hook).
